cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ZooKeeperSpecChecker.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model;
+
+import io.strimzi.api.kafka.model.status.Condition;
+import io.strimzi.operator.common.operator.resource.StatusUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Checks for potential problems in the configuration requested by the user, to provide
+ * warnings and share best practice. The intent is this class will generate warnings about
+ * configurations that aren't necessarily illegal or invalid, but that could potentially
+ * lead to problems.
+ */
+public class ZooKeeperSpecChecker {
+    private final ZookeeperCluster zoo;
+
+    /**
+     * @param zoo The model generated based on the spec. This is requested so that default
+     *                     values not included in the spec can be taken into account, without needing
+     *                     this class to include awareness of what defaults are applied.
+     */
+    public ZooKeeperSpecChecker(ZookeeperCluster zoo) {
```

Comment:
maybe we are using `zk` somewhere else instead of `zoo`

Reply:
I think I used `zoo` on most new things I wrote here. I can rename them to `zk` if you want?